content/post/053-js-ipfs-0.37.md

```diff
@@ -0,0 +1,216 @@
+---
+date: 2019-08-06
+url: 2019-08-06-js-ipfs-0-37
+title: js-ipfs 0.37.0 released
+author: Alan Shaw
+---
+
+> New constructor, better DNSLink support and delegated routing!
+
+# 🔦 Highlights
+
+## 👷‍♂️ Brand new constructor
+
+We're in the middle of [converting IPFS, libp2p and IPLD modules to using async/await and async iterators](https://github.com/ipfs/js-ipfs/issues/1670) instead of callbacks. This will eventually bubble up to the JS IPFS programmatic API where there will likely be some changes that will effect your application. To ease the transition when that happens we're encouraging new and existing developers away from using our callback based APIs, towards Promise based APIs. You'll see the change reflected in our API docs soon. For now, there's no breaking changes, all existing APIs continue to work as usual with callbacks.
+
+In this release there's a new way to construct an IPFS node using promises:
+
+```js
+const IPFS = require('ipfs')
+const node = await IPFS.create()
+// Done, ready to use!
+```
+
+All the usual options you'd normally pass to the constructor can be passed to `create`. Did we mention that it's completely backwards compatible? Well yeah, it is. No breaking changes here 😜 - you can still use `new IPFS()` and wait for the `ready` event as you used to. Or, alternatively, you can still use the constructor and `await` on the new "ready" promise like so:
+
+```js
+const IPFS = require('ipfs')
+const node = new IPFS()
+await node.ready
+// Done, ready to use!
+```
+
+More details in the [IPFS constructor docs](https://github.com/ipfs/js-ipfs#ipfs-constructor).
+
+## 🌎 Support for DNSLink IPNS name resolution
+
+JS IPFS can finally resolve a [DNSLink](http://dnslink.io/) IPNS path. It means that `jsipfs name resolve /ipns/ipfs.io` will now query DNS for a `dnslink` TXT record and return `/ipfs/QmRq5rhjnfFHYFYbYXoqPpcJQHKu3SUuHADzSzX9ECN5eM` (for example).
+
+It also works recursively, so your DNSLink TXT record could return another IPNS path to resolve and it would keep going. Turtles all the way down 🐢. Cowabunga!
+
+Bonus 🎁 this also works on the gateway so `http://127.0.0.1:8080/ipns/ipfs.io` will resolve and display the ipfs.io website. Hooray 😁!
+
+## 🧭 Delegated peer and content routing
+
+JS IPFS now supports delegating peer and content routing to another node on the network.
+
+What does that mean though? Well, when you delegate, you get someone else to do something. In delegated peer and content routing we get a different IPFS node to find a piece of content or another peer on the network.
+
+Delegation allows JS IPFS to make use of the routing abilities of other nodes. Typically we delegate to IPFS nodes that have unrestricted access to a DHT. This is useful for IPFS nodes running on Node.js, in the browser, or even for nodes running in resource constrained environments.
```

Comment:
Should this be a bit more descriptive for node? Right now this paragraph makes me think I should be using delegate routers regardless of my JS setup. Perhaps we should also add a note about delegates being used as fallbacks when the DHT is running?

```suggestion
Delegation allows JS IPFS to make use of the routing abilities of other nodes. Typically we delegate to IPFS nodes that have unrestricted access to a DHT. This is useful for IPFS nodes running in the browser, or even for nodes running in resource constrained environments. Node.js nodes running behind NATs or that are not running a DHT will also benefit from leveraging delegate routers. If you are running a DHT, the delegate will be used as a fallback router.
```

Reply:
Nice, yes. Good suggestion. I'm going to tweak "Node.js nodes" a little because too many "node" 😂